src/hooks/useLoader.ts

```diff
@@ -86,9 +91,24 @@ export function useLoader<T>(
 
   useEffect(() => {
     const task = run(function* () {
-      yield* setUpdateContext(setState);
-      yield* update({ type: "initial" })
-      yield* loader();
+      const channel = createChannel<LoaderState<unknown>>();
+
+      yield* setUpdateContext(channel.send);
+
+      const updates = yield* spawn(function*() {
+        for (const value of yield* each(channel)) {
+          setState(value);
+          if (value.type === "stopped") {
+            throw new Error("Forced interrupt")
```

Comment:
You can just stop your iteration and the updates task will win your race. I'd do that since it is an expected condition.
```suggestion
            break;
```

Reply:
I'm throwing the error to short-circuit the other child operations. Will break do the same here?

Reply:
If I read it correctly, then yes. breaking here exits the loop, which then causes the `updates` task to win the race, which then causes the main task invoked with `run()` to return, which then shuts down everything created in that scope.

Reply:
oh nice, I didn't catch that.